### ISSUE TITLE
StepTimer#mean should work regardless of other calls

### ISSUE THREAD

**Issue by crankydillo** [micrometer-metrics/micrometer#1814]
`StepTimer#mean` uses `Timer` [default implementation](https://github.com/micrometer-metrics/micrometer/blob/9c808373d59e7ae91b9873272fe9e30bb5bbd3cb/micrometer-core/src/main/java/io/micrometer/core/instrument/Timer.java#L186).  The problem with this is `count() == 0` will short-circuit the 'rolling' of `totalTime` at step intervals (unless some other code calls `totalTime`).  This is demonstrated by the following test:

```
    @Test
    void meanShouldWorkIfTotalTimeNotCalled() {
        final Duration stepDuration = Duration.ofMillis(10);
        final MockClock clock = new MockClock();
        final StepTimer timer = new StepTimer(
            mock(Meter.Id.class),
            clock,
            DistributionStatisticConfig.builder().expiry(stepDuration).bufferLength(2).build(),
            mock(PauseDetector.class),
            TimeUnit.MILLISECONDS,
            stepDuration.toMillis(),
            false
        );

        clock.add(Duration.ofMillis(10));
        assertThat(timer.mean(TimeUnit.MILLISECONDS)).isEqualTo(0.0);
        clock.add(Duration.ofMillis(1));
        timer.record(Duration.ofMillis(50));
        timer.record(Duration.ofMillis(100));
        clock.add(Duration.ofMillis(9));
        assertThat(timer.mean(TimeUnit.MILLISECONDS)).isEqualTo(75.0);
    }
```
If you add `timer.totalTime(TimeUnit.NANOSECONDS);` after the first assertion, the test will pass.
We use a custom `StepMeterRegistry` and are currently getting around this by overriding `Timer#mean` and calling `totalTime` before calling `super.mean`.

**Comment by crankydillo** [comment]
I almost submitted a PR to `StepTimer` with `mean` defined as so:
```
    @Override
    public double mean(final TimeUnit unit) {
        final double totalTime = totalTime(unit);
        final long count = count();
        return count == 0 ? 0.0 : totalTime / (double)count;
    }
```
That would solve my problem; however, I fear it is skirting the bigger issue, which is that for these 'Step' member variables to work correctly, they all need to be called at each step to make sure that their values 'roll' correctly.  Is that right or am I missing something?  Should we roll everything (i.e. `max`, `count`, and `total`) if any of them are polled?

**Comment by jkschneider** [comment]
@crankydillo Introducing a `StepTuple2<T1, T2>` so we can synchronize this rolling over two variables.

**Comment by jkschneider** [comment]
Merged forward to master with d26f4068f29f83bad95f8c9d1d0a1c358a439136.
